src/protocol.h

```diff
@@ -72,8 +72,8 @@ class Protocol : public std::enable_shared_from_this<Protocol>
 		void enableXTEAEncryption() {
 			encryptionEnabled = true;
 		}
-		void setXTEAKey(xtea::key key) {
-			this->key = std::move(key);
+		void setXTEAKey(const xtea::key& key) {
+			xtea::precomputeKeys(std::move(key), this->key);
```

Comment:
Please do not use C-style output parameters. It should read

```
this->key = xtea::precomputeKeys(key);
```

Reply:
@ranisalt which c++ stl header should i pick for that? i actually did tried that before but c++ std::arrays didnt seen flexible enough for this, it has no operator=, and because we already use c-style arrays in encryption (data, len arguments) i infered that it wouldn't be a problem. i did also considered a std::vector, but i couldn't tell by looking at it if that would make an allocation, and i don't think it would be good to have a dynamic array thing for this, since its fixed and should be

Reply:
`std::array` does have an equality operator, and it is the correct data type, just increase it to 64 items

Reply:
> because we already use c-style arrays in

By the way, this is completely different than using output pointer. We need to use pointers to avoid copies for performance, *not* to return information.

Reply:
@ranisalt derp, my bad, when i first did that one type i had in Protocol and xtea.h didn't match so the compiler said it didn't found an overloaded operator=, i assume std::array didn't had one.

Should i replace roundKeys with xtea::key ? i didn't made that change so there is an explicit distinction what is the key we got from the network, and what is the precomputed round keys